libgraph/include/katana/GraphTopology.h

```diff
@@ -1595,6 +1598,30 @@ class KATANA_EXPORT EdgeTypeAwareBiDirTopology
   }
 };
 
+template <typename Graph>
+using has_undirected_t =
+    decltype(std::declval<Graph>().UndirectedEdges(typename Graph::Node()));
+
+template <typename Graph, typename GNode = typename Graph::Node>
+auto
+OutOrUndirectedEdges(const Graph& graph, GNode node) {
+  if constexpr (katana::is_detected_v<has_undirected_t, Graph>) {
+    return graph.UndirectedEdges(node);
```

Comment:
Does `constexpr if` allow `SFINAE`?  e.g., will this work if some `graph` didn't support `UndirectedEdges`?

Reply:
Yep, this branch will be selected at compile time *only* if `graph.UndirectedEdges(node)` is valid.